internal/action/command.go

```diff
@@ -732,58 +732,56 @@ func (h *BufPane) SetLocalCmd(args []string) {
 	}
 }
 
-// ToggleCmd toggles an option
+// ToggleCmd toggles a boolean option
 func (h *BufPane) ToggleCmd(args []string) {
 	if len(args) < 1 {
-		InfoBar.Error("Please provide an option to toggle")
+		InfoBar.Error("Not enough arguments")
 		return
 	}
-	
+
 	option := args[0]
 	curVal := config.GetGlobalOption(option)
 	if curVal == nil {
 		InfoBar.Error(config.ErrInvalidOption)
 		return
 	}
-	
-	optionKind := reflect.TypeOf(curVal).Kind()
-	if optionKind != reflect.Bool {
-		InfoBar.Error("Non-boolean option")
+
+	curValBool, ok := curVal.(bool)
+	if !ok {
+		InfoBar.Error("Not a boolean option")
 		return
 	}
 	
-	newVal := strconv.FormatBool(!curVal.(bool))
-	err := SetGlobalOption(option, newVal)
+	err := SetGlobalOptionNative(option, !curValBool)
 	if err == config.ErrInvalidOption {
-		err := h.Buf.SetOption(option, newVal)
+		err := h.Buf.SetOptionNative(option, !curValBool)
 		if err != nil {
 			InfoBar.Error(err)
 		}
 	}
 }
 
-// ToggleLocalCmd toggles an option local to the buffer
+// ToggleLocalCmd toggles a boolean option local to the buffer
 func (h *BufPane) ToggleLocalCmd(args []string) {
 	if len(args) < 1 {
 		InfoBar.Error("Not enough arguments")
 		return
 	}
 	
 	option := args[0]
-	curVal := h.Buf.Settings[option]
+	curVal := config.GetGlobalOption(option)
```

Comment:
I think it's better to use `h.Buf.Settings[option]`, not just to avoid confusion when reading the code (although I understand the cause of this confusion is the confusing name of `GetGlobalOption()`, not your code), but also to check if this not a global-only option. With your current version, `togglelocal infobar` silently "succeeds" (and then `show infobar` meaninglessly says that infobar is false for this buffer). In comparison,  `setlocal infobar off` correctly fails and shows an error message (albeit perhaps not the clearest one).